core/src/main/java/tc/oc/pgm/filters/FilterParser.java

```diff
@@ -418,6 +418,31 @@ public GoalFilter parseGoal(Element el) throws InvalidXMLException {
     return new GoalFilter(goal, team, anyTeam);
   }
 
+  @MethodParser("completed")
+  public GoalFilter parseCompleted(Element el) throws InvalidXMLException {
+    XMLFeatureReference<? extends GoalDefinition> goal =
+        this.factory.getFeatures().createReference(new Node(el), GoalDefinition.class);
+
+    return new GoalFilter(goal, null, true);
+  }
+
+  @MethodParser("captured")
+  public GoalFilter parseCaptured(Element el) throws InvalidXMLException {
+    XMLFeatureReference<? extends GoalDefinition> goal =
+        this.factory.getFeatures().createReference(new Node(el), GoalDefinition.class);
+
+    Attribute attrTeam = el.getAttribute("team");
+    XMLFeatureReference<TeamFactory> team;
+    if (attrTeam != null) {
+      team = this.factory.getFeatures().createReference(new Node(attrTeam), TeamFactory.class);
+
+    } else {
+      team = null;
+    }
```

Comment:
```suggestion
    XMLFeatureReference<TeamFactory> team = attrTeam != null ? this.factory.getFeatures().createReference(new Node(attrTeam), TeamFactory.class) : null;
```
Just a personal opinion on format. Good work 👍

Reply:
yea that's a much cleaner way of doing it, thanks for the feedback